### ISSUE TITLE
Model type and training

### ISSUE THREAD

**Issue by IanTayler** [IanTayler/tenc#14]
TenC needs an easy way to define a model and later train it. Something that includes the possibility of setting up (and reusing) layers, defining loss functions, and training through stochastic gradient descent.
